mpmath/tests/test_convert.py

```diff
@@ -254,7 +254,9 @@ def test_compatibility():
         # Handle the weird types
         try: diff = np.abs(type(np.sqrt(x))(sqrt(x)) - np.sqrt(x))
         except: continue
-        assert diff < 2.0**-53
+        # numpy-2 does the comparison on LHS type, so we need to convert
+        # it to a type that can actually represent 2.0**-53
+        assert np.float64(diff) < 2.0**-53
```

Comment:
Shouldn't you cast the second operand instead?
```suggestion
        assert diff < np.float64(2.0**-53)
```

> numpy-2 does the comparison on LHS type

Could you point to docs?  This is unusual wrt the CPython behaviour.

Reply:
> Shouldn't you cast the second operand instead?

Oh yes, that makes more sense. Done that.

> > numpy-2 does the comparison on LHS type
> 
> Could you point to docs? This is unusual wrt the CPython behaviour.

It's just my guess. Couldn't find anything specific in the changelog or docs. Do you want me to report it to numpy, in case it wasn't an intentional change?

Reply:
> It's just my guess.

On which ground?

I would expect, that rich comparison will behave like it's specified in the Python language docs:
https://docs.python.org/3/reference/datamodel.html#object.__lt__

Else, numpy should document differences.

PS: please avoid force pushes.  Just add new commits.

Reply:
> On which ground?

Err, the results? Though I guess it's really more like "given numpy dtype and Python float, convert to numpy dtype".